src/main/java/edu/harvard/iq/dataverse/PermissionServiceBean.java

```diff
@@ -750,6 +750,21 @@ else if (dataset.isLockedFor(DatasetLock.Reason.InReview)) {
             }
         }
     }
+
+    public void checkUpdateDatasetVersionLock(Dataset dataset, DataverseRequest dataverseRequest, Command command) throws IllegalCommandException {
+        boolean locked = false;
+        if (dataset.isLocked()) {
+            for (final DatasetLock lock: dataset.getLocks()) {
+                if (lock.getReason() != DatasetLock.Reason.Ingest) {
+                    locked = true;
+                    break;
+                }
+            }
+        }
+        if (locked) {
+            checkEditDatasetLock(dataset, dataverseRequest, command);
+        }
+    }
```

Comment:
Not sure I understand this  - the code only calls checkEditDatasetLock if there is an ingest lock, but that call will immediately throw an IllegalCommandException due to the Ingest lock.

Reply:
The goal here is to call checkEditDatasetLock only if there is at least one lock that is not an ingest lock. If there is no lock, or there are only ingest locks, it is fine to do an update. Otherwise, default checkEditDatasetLock is called.

Reply:
I have rewritten the code for more readability:
```
boolean hasAtLeastOneLockThatIsNotAnIngestLock = dataset.isLocked() && dataset.getLocks().stream()
    .anyMatch(lock -> !DatasetLock.Reason.Ingest.equals(lock.getReason()));
if (hasAtLeastOneLockThatIsNotAnIngestLock) {
    checkEditDatasetLock(dataset, dataverseRequest, command);
}
```